charts/presto/templates/presto-common-config.yaml

```diff
@@ -5,7 +5,56 @@ metadata:
 {{- block "extraMetadata" . }}
 {{- end }}
 data:
-  environment: {{ .Values.spec.presto.config.environment | quote }}
-  discovery-uri: {{ .Values.spec.presto.config.discoveryURI | quote }}
-  hive-metastore-uri: {{ .Values.spec.presto.config.hiveMetastoreURI }}
+  entrypoint.sh: |
```

Comment:
I'd prefer to do it in the entrypoint since I'd rather not have the initContainer in the first place. The initContainer is hack because downwardAPI volumes are mounted readOnly. If we could get rid of that, the entrypoint will still just work, whereas if we made it all initContainer oriented, when removing the initContainer, we would have to rewrite the entrypoint script.

Reply:
Is there a timeline or even a ticket/issue anywhere to indicate that we might be able to modify things on Downward API volumes in the future? It doesn't seem particularly likely that that'd ever be supported in Kubernetes, and since we're already building this explicitly around that, I just feel like it might be cleaner to put all of the configuration initialization in the init volume. 

Ultimately I'm not majorly attached to this idea, so I won't block your PR on it, I just want to make sure it's given its full consideration :)

Reply:
I'm not sure. Overall, I hope there's a better way than using init-containers in the future, and the other benefit of the entrypoint is it works with, or without any changes so I still find the design appealing, versus an initContainer specific approach.

However, we could in theory runn the entrypoint script in the initContainer, and then just running presto directly in the other container. The bigger pain is we would need to copy /etc/passwd around in addition, since that part of the logic isn't contained in the files being copied now.

Reply:
Yeah, looking at it more, handling /etc/passwd probably always needs to be done in the entrypoint script so that convinces me more that this is probably the best for now.